weatherapp/src/App.js

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useContext } from 'react';
+import logo from './logo.svg';
+import { Route, Switch } from "react-router-dom";
+import './App.css';
+import WeatherApp from "./components/weatherHome"
+import Home from "./components/home"
+import NavBar from './components/NavBar';
+import DarkContext from "./providers/darkModeProvider"
+
+function App() {
+  const [ darkMode, setDarkMode] = useState(false)
```

Comment:
You want to create a Context provider for this darkmode and lightmode, this will help you so u dont have to pass down props through all the children components